# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: gfx/gl/GLBlitHelper.cpp
# Commit: ccfa767dba64
# Full Hash: ccfa767dba644dc193e0246eb8e8ff3377e8b8a5
# Author: Jeff Gilbert <jgilbert@mozilla.com>
# Date: 2020-01-09 03:40:18
# Regressor Bug: 1477756
# File Overlap Count: 1
# Description:
#   Bug 1477756 - Client-side bindings mirror for precise CC, and merge similar codepaths. r=handyman
#   
#   * Context loss using RAII
#   * Move Program reflection Client-side
#   
# ==============================================================================

diff -r 71c122ac0ca7 -r ccfa767dba64 gfx/gl/GLBlitHelper.cpp
--- a/gfx/gl/GLBlitHelper.cpp	Wed Jan 08 22:19:14 2020 +0000
+++ b/gfx/gl/GLBlitHelper.cpp	Wed Jan 08 22:19:16 2020 +0000
@@ -684,15 +684,23 @@
       return BlitImage(static_cast<PlanarYCbCrImage*>(srcImage), destSize,
                        destOrigin);
 
+    case ImageFormat::SURFACE_TEXTURE:
 #ifdef MOZ_WIDGET_ANDROID
-    case ImageFormat::SURFACE_TEXTURE:
       return BlitImage(static_cast<layers::SurfaceTextureImage*>(srcImage),
                        destSize, destOrigin);
+#else
+      MOZ_ASSERT(false);
+      return false;
 #endif
-#ifdef XP_MACOSX
+
     case ImageFormat::MAC_IOSURFACE:
+#ifdef XP_MACOSX
       return BlitImage(srcImage->AsMacIOSurfaceImage(), destSize, destOrigin);
+#else
+      MOZ_ASSERT(false);
+      return false;
 #endif
+
 #ifdef XP_WIN
     case ImageFormat::GPU_VIDEO:
       return BlitImage(static_cast<layers::GPUVideoImage*>(srcImage), destSize,
@@ -705,11 +713,21 @@
                        destSize, destOrigin);
     case ImageFormat::D3D9_RGB32_TEXTURE:
       return false;  // todo
+#else
+    case ImageFormat::GPU_VIDEO:
+    case ImageFormat::D3D11_SHARE_HANDLE_TEXTURE:
+    case ImageFormat::D3D11_YCBCR_IMAGE:
+    case ImageFormat::D3D9_RGB32_TEXTURE:
+      MOZ_ASSERT(false);
+      return false;
 #endif
-    default:
-      gfxCriticalError() << "Unhandled srcImage->GetFormat(): "
-                         << uint32_t(srcImage->GetFormat());
-      return false;
+
+    case ImageFormat::CAIRO_SURFACE:
+    case ImageFormat::NV_IMAGE:
+    case ImageFormat::OVERLAY_IMAGE:
+    case ImageFormat::SHARED_RGB:
+    case ImageFormat::TEXTURE_WRAPPER:
+      return false;  // todo
   }
 }
 